### PR TITLE
Keithley smu 4200scs fix

### PR DESCRIPTION
**Pull request by afischer-sweepme**
Keithley 4200-SCS changes to make GPIB connection via KXCI server working again.
lpt.initialize() in try-except to throw a better error message

### CODE REVIEW

File: src/SMU-Keithley_4200-SCS/main.py [hunk 1]
```diff
@@ -236,8 +235,13 @@ def connect(self):
...
                 self.param = param
 
-            ret = self.lpt.initialize()
-            
+            try:
+                ret = self.lpt.initialize()
+            except ConnectionRefusedError as e:
+                debug("Unable to connect to a lptlib server application running on the 4200-SCS. Please check your"
+                      "network settings and make sure the server application is running.")
+                raise e
```
Comment (fk3) [inline src/SMU-Keithley_4200-SCS/main.py:243]:
```suggestion
                raise
```
In the except: block a bare `raise` already raises the currently handled exception. This also makes the traceback clearer as it does not contain the rather confusing `raise e in line 243`

File: src/SMU-Keithley_4200-SCS/main.py [hunk 2]
```diff
@@ -372,7 +376,7 @@ def configure(self):
                 # Current Range
                 if "Limited" in self.current_range:
                     current_range = self.current_ranges[self.current_range]
-                    self.set_current_range_limited(self.card_name, current_range)  # low range current
+                    self.set_current_range_limited(self.card_name[-1], current_range)  # low range current
```
Comment (fk3) [inline src/SMU-Keithley_4200-SCS/main.py:379]: I tried to figure out how and why this fixes issues, but it seems quite difficult to understand what self.card_name is or it seems to be very different for various scenarios (sometimes the channel, sometimes a concatenation with the channel, sometimes a split...). An explanation when card_name is first defined would be quite useful.

File: src/Signal-Keithley_4200-SCS/main.py [hunk 3]
```diff
@@ -198,7 +197,12 @@ def connect(self):
                 # not supported yet with pylptlib -> # self.inst = inst
                 self.param = param
 
-            ret = self.lpt.initialize()
+            try:
+                ret = self.lpt.initialize()
+            except ConnectionRefusedError as e:
+                debug("Unable to connect to a lptlib server application running on the 4200-SCS. Please check your"
+                      "network settings and make sure the server application is running.")
```
Comment (fk3) [inline src/Signal-Keithley_4200-SCS/main.py:204]:
```suggestion
                debug("Unable to connect to a lptlib server application running on the 4200-SCS. Please check your "
                      "network settings and make sure the server application is running.")
```
space required, I think

File: src/Signal-Keithley_4200-SCS/main.py [hunk 4]
```diff
@@ -198,7 +197,12 @@ def connect(self):
...
+                raise e
```
Comment (fk3) [inline src/Signal-Keithley_4200-SCS/main.py:205]:
```suggestion
                raise
```
Same explanation as for the SMU driver

File: src/SMU-Keithley_4200-SCS/main.py (hunk 2 above)
Comment (afischer-sweepme) [reply src/SMU-Keithley_4200-SCS/main.py:385]: added a description of self.card_name

File: src/SMU-Keithley_4200-SCS/main.py [hunk 5]
```diff
@@ -372,7 +376,7 @@ def configure(self):
...
 
             # range = 1e-1
             # compliance = 1e1
```
[the remark this answers is not in the document]
Comment (afischer-sweepme) [reply src/SMU-Keithley_4200-SCS/main.py]: Add a commented that this needs to be done when working with the instrument again.

**Review by fk3 — APPROVED** [review]

**Pull request merged**